device/src/kernel/actor.rs

```diff
@@ -1,31 +1,28 @@
 use super::{
-    channel::{Channel, ChannelError, ChannelReceive, ChannelReceiver, ChannelSender},
     signal::{SignalFuture, SignalSlot},
     util::ImmediateFuture,
 };
 use core::cell::{RefCell, UnsafeCell};
 use core::future::Future;
 use core::pin::Pin;
 use core::task::{Context, Poll};
-use embassy::executor::{raw::Task, SpawnError, SpawnToken, Spawner};
-use embassy::util::DropBomb;
-use generic_array::GenericArray;
-use heapless::{consts, ArrayLength};
+
+use embassy::{
+    executor::{raw::Task, SpawnError, SpawnToken, Spawner},
+    util::{
+        mpsc::{self, Channel, Receiver, RecvFuture, Sender, WithCriticalSections},
+        DropBomb,
+    },
+};
+
+type ActorMutex = WithCriticalSections;
```

Comment:
I’m sure you’ve thought of this, but there’s a NoopMutex available also, which is probably more typical to use with Embassy.

Reply:
Ah, indeed. Thanks for the suggestion, that fits better.